trace/trace.go

```diff
@@ -123,34 +130,82 @@ type StartOptions struct {
 	// SpanKind represents the kind of a span. If none is set,
 	// SpanKindUnspecified is used.
 	SpanKind int
+
+	remoteParent SpanContext
+}
+
+// StartOption apply changes to StartOptions.
+type StartOption func(*StartOptions)
+
+// WithSpanKind makes new spans to be created with the given kind.
+func WithSpanKind(spanKind int) StartOption {
+	return func(o *StartOptions) {
+		o.SpanKind = spanKind
+	}
+}
+
+// WithSampler makes new spans to be be created with a custom sampler.
+// Otherwise, the global sampler is used.
+func WithSampler(sampler Sampler) StartOption {
+	return func(o *StartOptions) {
+		o.Sampler = sampler
+	}
+}
+
+// WithRemoteParent makes new spans to be the child of the given part.
+//
+// WithRemoteParent is used to set a parent that has been
+// propagated in an incoming request or RPC.
+//
+// If there is already a parent in the incoming context.Context,
+// this option replaces it with the given parent.
+func WithRemoteParent(parent SpanContext) StartOption {
```

Comment:
I think this is not actually an "optional" thing, in case of a grpc/http server plugins this is required to be set and in all the other cases this is required to not be set. I will have 2 apis StartSpan and StartSpanWithRemoteParent and make this a required field in the second API.

Reply:
I see your point. I added StartSpanWithRemoteParent and removed the WithRemoteParent.